### DISCUSSION TITLE
Default application font size?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#9950] [General]
I'm using a 4K monitor (on Linux) and the default font size for Avalonia apps is 12 (SimpleTheme / 11.0.0-preview4). This font size appears tiny and Avalonia doesn't seem to recognise default interface fonts or sizes.

Is there a way to set the default window font size in an application? And how would I read this font size in code?

**Comment by kuiperzone** [comment]
Alternatively, is there some way for handling high-res monitors and scaling appropriately?

**Comment by maxkatz6** [comment]
Default font size can be changed by chaning Window.FontSize or any other top-level control you want. New value will be inherited down the tree.

But in your case, there is another problem with scaling it seems.
These issues might help you https://github.com/AvaloniaUI/Avalonia/issues/9390 https://github.com/AvaloniaUI/Avalonia/issues/4826

**Reply by kuiperzone** [reply]
Thanks. Indeed some of the issues you give are relevant. Am running Wayland BTW.

So, yes, I can set the Window.FontSize for every window. But there is no App.DefaultFontSize?

Also, the only way I can find to programmatically detect Window.FontSize is to create a dummy window and read FontSize?

**Reply by maxkatz6** [reply]
Default value for the window font size is read from the resources. At least with our default Fluent theme. 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Accents/Base.xaml#L16

You can read and change this resource by accessing it from Application.Current.Resources (indexer to set or TryFindResource to read)

**Reply by kuiperzone** [reply]
Thank you for reply. Was helpful.

PS. Looking forward to version 0.11. Great work!
